Error compiling objects for extension while installing with python

The error log is as follows.
`FAILED: /home/liuchang/MinkowskiEngine/build/temp.linux-x86_64-3.9/home/liuchang/MinkowskiEngine/src/spmm.o 
/usr/local/cuda/bin/nvcc  -I/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/include -I/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/include/torch/csrc/api/include -I/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/include/TH -I/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/liuchang/MinkowskiEngine/src -I/home/liuchang/MinkowskiEngine/src/3rdparty -I/home/liuchang/anaconda3/envs/PCGCv2/include/python3.9 -c -c /home/liuchang/MinkowskiEngine/src/spmm.cu -o /home/liuchang/MinkowskiEngine/build/temp.linux-x86_64-3.9/home/liuchang/MinkowskiEngine/src/spmm.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' --expt-relaxed-constexpr --expt-extended-lambda -O3 -Xcompiler=-fno-gnu-unique -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1011"' -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -gencode=arch=compute_61,code=compute_61 -gencode=arch=compute_61,code=sm_61 -std=c++14
/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/include/c10/util/ArrayRef.h(142): error: expression must be a pointer to a complete object type
          detected during instantiation of "c10::ArrayRef<T>::iterator c10::ArrayRef<T>::end() const [with T=at::Tensor]" 
/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/include/ATen/cuda/CUDAUtils.h(14): here

/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/include/ATen/cuda/CUDAUtils.h(15): error: incomplete type is not allowed

/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/include/ATen/NamedTensorUtils.h(14): error: no instance of function template "std::any_of" matches the argument list
            argument types are: (const at::Tensor *, <error-type>, lambda [](const at::Tensor &)->__nv_bool)

/home/liuchang/MinkowskiEngine/src/spmm.cu(102): warning: variable "is_int64" was declared but never referenced

/home/liuchang/MinkowskiEngine/src/spmm.cu(364): warning: variable "is_int64" was declared but never referenced

/home/liuchang/MinkowskiEngine/src/spmm.cu(473): warning: variable "num_unique_keys" was set but never used
          detected during instantiation of "std::vector<at::Tensor, std::allocator<at::Tensor>> minkowski::coo_spmm_average<th_int_type>(const at::Tensor &, const at::Tensor &, int64_t, int64_t, const at::Tensor &, int64_t) [with th_int_type=int32_t]" 
(601): here

/home/liuchang/MinkowskiEngine/src/spmm.cu(473): warning: variable "num_unique_keys" was set but never used
          detected during instantiation of "std::vector<at::Tensor, std::allocator<at::Tensor>> minkowski::coo_spmm_average<th_int_type>(const at::Tensor &, const at::Tensor &, int64_t, int64_t, const at::Tensor &, int64_t) [with th_int_type=int32_t]" 
(601): here

3 errors detected in the compilation of "/tmp/tmpxft_000046e4_00000000-6_spmm.cpp1.ii".
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 1740, in _run_ninja_build
    subprocess.run(
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['ninja', '-v', '-j', '1']' returned non-zero exit status 1.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/liuchang/MinkowskiEngine/setup.py", line 317, in <module>
    setup(
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/setuptools/__init__.py", line 153, in setup
    return distutils.core.setup(**attrs)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/setuptools/command/install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/setuptools/command/bdist_egg.py", line 164, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/setuptools/command/bdist_egg.py", line 150, in call_command
    self.run_command(cmdname)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/setuptools/command/install_lib.py", line 11, in run
    self.build()
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/command/install_lib.py", line 107, in build
    self.run_command('build_ext')
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/setuptools/command/build_ext.py", line 79, in run
    _build_ext.run(self)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/command/build_ext.py", line 340, in run
    self.build_extensions()
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 741, in build_extensions
    build_ext.build_extensions(self)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/command/build_ext.py", line 449, in build_extensions
    self._build_extensions_serial()
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/command/build_ext.py", line 474, in _build_extensions_serial
    self.build_extension(ext)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/setuptools/command/build_ext.py", line 202, in build_extension
    _build_ext.build_extension(self, ext)
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/distutils/command/build_ext.py", line 529, in build_extension
    objects = self.compiler.compile(sources,
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 562, in unix_wrap_ninja_compile
    _write_ninja_file_and_compile_objects(
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 1419, in _write_ninja_file_and_compile_objects
    _run_ninja_build(
  File "/home/liuchang/anaconda3/envs/PCGCv2/lib/python3.9/site-packages/torch/utils/cpp_extension.py", line 1756, in _run_ninja_build
    raise RuntimeError(message) from e
RuntimeError: Error compiling objects for extension`

Have anyone faced the same problem while installing? Please tell me how to fix this. Thanks!

--- a/src/spmm.cu
+++ b/src/spmm.cu
@@ -29,6 +29,7 @@
 
 #include <cusparse.h>
 
+#include <ATen/core/Tensor.h>
 #include <ATen/cuda/CUDAContext.h>
 #include <ATen/cuda/CUDAUtils.h>
 #include <c10/cuda/CUDACachingAllocator.h>

